[Bug] client.start()가 작동하지 않습니다

## Summary<br/>
> 본 버그 제보는 Developer Space 공개 이슈로 제보된 이슈입니다.

client.run()은 작동 하는데 이유는 모르겠지만 client.start()는 안되네요.

동작한 소스코드는 다음과 같습니다.
```py
import discord, os, json, asyncio, logging, data
from discord.ext import commands
from chzzkpy import Client, Donation, Message, UserPermission, unofficial

DEBUG = False

logging.basicConfig(
    level=logging.DEBUG if DEBUG else logging.INFO,
    format="[%(levelname)s] %(message)s",
    handlers=[
        logging.FileHandler(f"{__name__}.log", encoding="utf-8"),
        logging.StreamHandler()
    ]
)

logger = logging.getLogger(__name__)

# setting
file_path = os.path.dirname(os.path.abspath(__file__))
data_mod = "test"

# Libee discord server settings
verify_role_id = 0

# discord.py settings
use_cogs = [file.split(".")[0] for file in os.listdir(f"{file_path}/cogs") if ".py" in file]
command_prefix = "!"

# chzzkpy settings
chzzk_client = Client(data.CHZZK_OFFICIAL_CLIENT_ID, data.CHZZK_OFFICIAL_CLIENT_SECRET)
discord_link_client = Client(data.CHZZK_OFFICIAL_DISCORD_LINK_CLIENT_ID, data.CHZZK_OFFICIAL_DISCORD_LINK_CLIENT_SECRET)
chzzk_unofficial_client = unofficial.chat.ChatClient(data.LIBEE_CHZZK_ID)
song_list = []

# DISCORD API
class Bot(commands.Bot):
    async def setup_hook(self):
        with open(f"{file_path}/data/{data_mod}.json", "r", encoding="utf-8") as f:
            data = json.load(f)
        data["file_path"] = file_path
        bot.libee_data = data
        bot.libee_data["chzzk"] = discord_link_client
        bot.data_mod = data_mod
        bot.use_cogs = use_cogs

        for cog in use_cogs:
            await bot.load_extension(f'cogs.{cog}')
            print(f'Cog {cog} has been loaded.')
        try:
            synced = await bot.tree.sync()
            print(f"synced {len(synced)} commands")
        except Exception as e:
            print(e)
        print(f'{bot.user.name} has connected to Discord!')

bot = Bot(command_prefix=command_prefix, intents=discord.Intents.all())

@bot.command(name="start")
async def start(ctx):
    if ctx.author.id not in bot.libee_data["admins"]:
        await ctx.send("You don't have command permissions.")
        return
    result = []
    for name, cog in bot.cogs.items():
        if hasattr(cog, "main") and callable(getattr(cog, "main")):
            await cog.main(ctx)
            result.append(name)
    await ctx.send(f"Run {len(result)} commands. ({', '.join(result)})")

@bot.command(name="grant_role_perm")
async def grant_role_perm(ctx: commands.Context):
    guild = ctx.guild
    role = guild.get_role(1404943174060736592)
    if role is None:
        await ctx.send("❌ 해당 역할을 찾을 수 없습니다.")
        return

    # 예: 메시지 관리 권한 + 채널 관리 권한 부여
    perms = role.permissions
    perms.update(use_application_commands=True)

    await role.edit(permissions=perms)
    await ctx.send(f"✅ 역할 {role.mention} 에게 권한을 부여했습니다.")

# CHZZK API
async def format_response(content: str, message: unofficial.chat.Message):
    return content.format(
        user = message.profile.nickname,
        song_list = "\n".join(song_list),
        song_list_total = len(song_list))

@chzzk_unofficial_client.event
async def on_connect():
    logger.info("successfully connected to chzzk")

@chzzk_unofficial_client.event
async def on_chat(message: unofficial.chat.ChatMessage):
    message_split = message.content.split()
    if message_split[0] == "!노래신청":
        song_list.append(message_split[1:-1])
        await message.send(f"노래 신청됌 : {message_split[1:-1]}")
        return
    
async def main():
    bot_start = bot.start(data.DISCORD_BOT_TEST_TOKEN)
    chzzk_start = chzzk_unofficial_client.start(data.FAN_CLIENT_AUTHORIZATION_KEY, data.FAN_CLIENT_SESSION_KEY)
    bot_task = asyncio.create_task(bot_start)
    chzzk_task = asyncio.create_task(chzzk_start)
    await asyncio.gather(bot_task, chzzk_task)

if __name__ == "__main__":
    asyncio.run(main())
```

## Expected behavior<br/>
`unofficial.chat.ChatClient`가 가동되어 서버와 통신할 수 있어야 합니다.

## Actual Results<br/>
다음과 같이 예외가 발생합니다.
```pytb
[WARNING] PyNaCl is not installed, voice will NOT be supported
[INFO] logging in using static token
Traceback (most recent call last):
  File "c:\libee\main.py", line 114, in <module>
    asyncio.run(main())
    ~~~~~~~~~~~^^^^^^^^
  File "C:\Users\USER\AppData\Local\Programs\Python\Python313\Lib\asyncio\runners.py", line 195, in run
    return runner.run(main)
           ~~~~~~~~~~^^^^^^
  File "C:\Users\USER\AppData\Local\Programs\Python\Python313\Lib\asyncio\runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^
  File "C:\Users\USER\AppData\Local\Programs\Python\Python313\Lib\asyncio\base_events.py", line 725, in run_until_complete
    return future.result()
           ~~~~~~~~~~~~~^^
  File "c:\libee\main.py", line 111, in main
    await asyncio.gather(bot_task, chzzk_task)
  File "C:\Users\USER\AppData\Local\Programs\Python\Python313\Lib\site-packages\chzzkpy\unofficial\chat\chat_client.py", line 113, in start
    await self.connect()
  File "C:\Users\USER\AppData\Local\Programs\Python\Python313\Lib\site-packages\chzzkpy\unofficial\chat\chat_client.py", line 133, in connect
    status = await self.live_status(channel_id=self.channel_id)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\USER\AppData\Local\Programs\Python\Python313\Lib\site-packages\chzzkpy\unofficial\chat\chat_client.py", line 522, in live_status
    return await super().live_status(channel_id)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\USER\AppData\Local\Programs\Python\Python313\Lib\site-packages\chzzkpy\unofficial\client.py", line 130, in live_status
    res = await self._api_session.live_status(channel_id=channel_id)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\USER\AppData\Local\Programs\Python\Python313\Lib\site-packages\ahttp_client\request.py", line 580, in __call__
    response = await self.session._make_request(req_obj, formatted_path)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\USER\AppData\Local\Programs\Python\Python313\Lib\site-packages\ahttp_client\session.py", line 124, in _make_request
    response = await self.session.request(_req_obj.method, _path, **request_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\USER\AppData\Local\Programs\Python\Python313\Lib\site-packages\aiohttp\client.py", line 607, in _request
    with timer:
         ^^^^^
  File "C:\Users\USER\AppData\Local\Programs\Python\Python313\Lib\site-packages\aiohttp\helpers.py", line 636, in __enter__
    raise RuntimeError("Timeout context manager should be used inside a task")
RuntimeError: Timeout context manager should be used inside a task
[ERROR] Unclosed client session
client_session: <aiohttp.client.ClientSession object at 0x000001C1C4950050>
```
## Device Info<br/>
* Operation System: Windows 11
* Python version: v3.13.2
* chzzkpy version: v2.1.1-stable

## Additional context<br/>

--- a/chzzkpy/__init__.py
+++ b/chzzkpy/__init__.py
@@ -36,7 +36,7 @@ __title__ = "chzzkpy"
 __author__ = "gunyu1019"
 __license__ = "MIT"
 __copyright__ = "Copyright 2024-present gunyu1019"
-__version__ = "2.1.1"  # version_info.to_string()
+__version__ = "2.1.2"  # version_info.to_string()
 
 
 class VersionInfo(NamedTuple):
@@ -54,5 +54,5 @@ class VersionInfo(NamedTuple):
 
 
 version_info: VersionInfo = VersionInfo(
-    major=2, minor=1, micro=1, release_level=None, serial=0
+    major=2, minor=1, micro=2, release_level=None, serial=0
 )

--- a/chzzkpy/unofficial/chat/chat_client.py
+++ b/chzzkpy/unofficial/chat/chat_client.py
@@ -72,7 +72,10 @@ class ChatClient(Client):
         self.access_token: Optional[AccessToken] = None
         self.user_id: Optional[str] = None
 
-        self.ws_session = aiohttp.ClientSession(loop=self.loop)
+        self.ws_session = None
+
+        self.__authorization_key = authorization_key
+        self.__session_key = session_key
 
         self._listeners: dict[str, list[tuple[asyncio.Future, Callable[..., bool]]]] = (
             dict()
@@ -90,9 +93,17 @@ class ChatClient(Client):
         self._gateway: Optional[ChzzkWebSocket] = None
         self._status: Literal["OPEN", "CLOSE"] = None
 
-    def _session_initial_set(self):
-        self._api_session = ChzzkAPIChatSession(loop=self.loop)
-        self._game_session = NaverGameChatSession(loop=self.loop)
+    def _session_initial_set(self, loop: Optional[asyncio.AbstractEventLoop] = None):
+        self._api_session = ChzzkAPIChatSession(loop=self.loop or loop)
+        self._game_session = NaverGameChatSession(loop=self.loop or loop)
+
+        self.ws_session = aiohttp.ClientSession(loop=self.loop or loop)
+
+        if self.__authorization_key is not None and self.__session_key is not None:
+            self.login(self.__authorization_key, self.__session_key)
+
+        for manage_client in self._manage_client.values():
+            manage_client._session_initial_set(self.loop or loop)
 
     @property
     def is_connected(self) -> bool:
@@ -125,9 +136,14 @@ class ChatClient(Client):
         session_key : str
             A `NID_SES` value in the cookie.
         """
+        if self._api_session is None or self._game_session is None:
+            self.__authorization_key = authorization_key
+            self.__session_key = session_key
+
         super().login(authorization_key=authorization_key, session_key=session_key)
         self._manage_client[self.channel_id] = ManageClient(self.channel_id, self)
 
+    @Client.initial_async_setup
     async def connect(self) -> None:
         if self.chat_channel_id is None:
             status = await self.live_status(channel_id=self.channel_id)
@@ -161,7 +177,9 @@ class ChatClient(Client):
 
         if self._gateway is not None:
             await self._gateway.close()
-        await self.ws_session.close()
+
+        if self.ws_session is not None:
+            await self.ws_session.close()
         await super().close()
 
     async def _confirm_live_status(self):
@@ -194,6 +212,7 @@ class ChatClient(Client):
         self.chat_channel_id = live_status.chat_channel_id
         raise ReconnectWebsocket()
 
+    @Client.initial_async_setup
     async def polling(self) -> None:
         while not self.is_closed:
             try:
@@ -366,6 +385,7 @@ class ChatClient(Client):
         return self.loop.create_task(wrapped, name=f"chzzk.py: {event_name}")
 
     # API Method
+    @Client.initial_async_setup
     async def _generate_access_token(self) -> AccessToken:
         res = await self._game_session.chat_access_token(
             channel_id=self.chat_channel_id
@@ -374,6 +394,7 @@ class ChatClient(Client):
         return self.access_token
 
     # Chat Method
+    @Client.initial_async_setup
     async def send_chat(self, message: str) -> None:
         """Send a message.
 
@@ -395,6 +416,7 @@ class ChatClient(Client):
 
         await self._gateway.send_chat(message, self.chat_channel_id)
 
+    @Client.initial_async_setup
     async def request_recent_chat(self, count: int = 50):
         """Send a request recent chat to chzzk.
         This method only makes a “request”.
@@ -415,6 +437,7 @@ class ChatClient(Client):
 
         await self._gateway.request_recent_chat(count, self.chat_channel_id)
 
+    @Client.initial_async_setup
     async def history(self, count: int = 50) -> list[ChatMessage]:
         """Get messages the user has previously sent.
 
@@ -434,6 +457,7 @@ class ChatClient(Client):
         )
         return recent_chat.message_list
 
+    @Client.initial_async_setup
     async def set_notice_message(self, message: ChatMessage) -> None:
         """Set a pinned messsage.
 
@@ -456,11 +480,13 @@ class ChatClient(Client):
         )
         return
 
+    @Client.initial_async_setup
     async def delete_notice_message(self) -> None:
         """Delete a pinned message."""
         await self._game_session.delete_notice_message(channel_id=self.chat_channel_id)
         return
 
+    @Client.initial_async_setup
     async def blind_message(self, message: ChatMessage) -> None:
         """Blinds a chat.
 
@@ -478,6 +504,7 @@ class ChatClient(Client):
         )
         return
 
+    @Client.initial_async_setup
     async def temporary_restrict(self, user: PartialUser | str) -> PartialUser:
         """Give temporary restrict to user.
         A temporary restriction cannot be lifted arbitrarily.
@@ -504,7 +531,8 @@ class ChatClient(Client):
         )
         return response
 
-    async def live_status(self, channel_id: Optional[str] = None):
+    @Client.initial_async_setup
+    async def live_status(self, channel_id: Optional[str] = None) -> Optional[LiveStatus]:
         """Get a live status info of broadcaster.
 
         Parameters
@@ -521,6 +549,7 @@ class ChatClient(Client):
             channel_id = self.channel_id
         return await super().live_status(channel_id)
 
+    @Client.initial_async_setup
     async def live_detail(self, channel_id: Optional[str] = None):
         """Get a live detail info of broadcaster.
 
@@ -563,6 +592,7 @@ class ChatClient(Client):
         """Get a client provided self-channel management functionally."""
         return self.manage(channel_id=self.channel_id)
 
+    @Client.initial_async_setup
     async def profile_card(self, user: PartialUser | str) -> Profile:
         """Get a profile card.
 

--- a/chzzkpy/unofficial/chat/chat_client.py
+++ b/chzzkpy/unofficial/chat/chat_client.py
@@ -532,7 +532,9 @@ class ChatClient(Client):
         return response
 
     @Client.initial_async_setup
-    async def live_status(self, channel_id: Optional[str] = None) -> Optional[LiveStatus]:
+    async def live_status(
+        self, channel_id: Optional[str] = None
+    ) -> Optional[LiveStatus]:
         """Get a live status info of broadcaster.
 
         Parameters
@@ -550,7 +552,9 @@ class ChatClient(Client):
         return await super().live_status(channel_id)
 
     @Client.initial_async_setup
-    async def live_detail(self, channel_id: Optional[str] = None):
+    async def live_detail(
+        self, channel_id: Optional[str] = None
+    ) -> Optional[LiveDetail]:
         """Get a live detail info of broadcaster.
 
         Parameters

--- a/chzzkpy/unofficial/chat/http.py
+++ b/chzzkpy/unofficial/chat/http.py
@@ -87,8 +87,8 @@ class NaverGameChatSession(NaverGameAPISession):
         extras: Annotated[str, BodyJson],
         message: Annotated[str, BodyJson],
         message_time: Annotated[int, BodyJson.to_camel()],
-        message_user_id_hash: Annotated[int, BodyJson.to_camel()],
-        streaming_channel_id: Annotated[int, BodyJson.to_camel()],
+        message_user_id_hash: Annotated[str, BodyJson.to_camel()],
+        streaming_channel_id: Annotated[str, BodyJson.to_camel()],
         chat_type: Annotated[str, BodyJson.to_camel()] = "STREAMING",
     ) -> Content[None]:
         return
@@ -101,8 +101,8 @@ class NaverGameChatSession(NaverGameAPISession):
         channel_id: Annotated[str, BodyJson.to_camel()],
         message: Annotated[str, BodyJson],
         message_time: Annotated[int, BodyJson.to_camel()],
-        message_user_id_hash: Annotated[int, BodyJson.to_camel()],
-        streaming_channel_id: Annotated[int, BodyJson.to_camel()],
+        message_user_id_hash: Annotated[str, BodyJson.to_camel()],
+        streaming_channel_id: Annotated[str, BodyJson.to_camel()],
         chat_type: Annotated[str, BodyJson.to_camel()] = "STREAMING",
     ) -> Content[None]:
         pass

--- a/chzzkpy/unofficial/client.py
+++ b/chzzkpy/unofficial/client.py
@@ -26,11 +26,15 @@ from __future__ import annotations
 import asyncio
 
 from typing import Optional, TYPE_CHECKING
+from functools import wraps
+
 from .http import ChzzkAPISession, NaverGameAPISession
 from .live import Live, LiveStatus, LiveDetail
 from .user import User
 from .video import Video
 from .manage.manage_client import ManageClient
+
+from ..client import _LoopSentinel
 
 
 if TYPE_CHECKING:
@@ -48,25 +52,47 @@ class Client:
         authorization_key: Optional[str] = None,
         session_key: Optional[str] = None,
     ):
-        self.loop = loop or asyncio.get_event_loop()
+        self.loop = loop or _LoopSentinel()
         self._closed = False
         self._api_session = None
         self._game_session = None
 
-        if authorization_key is not None and session_key is not None:
-            self.login(authorization_key, session_key)
-
-        self._session_initial_set()
+        self.__authorization_key = authorization_key
+        self.__session_key = session_key
 
         # For management feature
         self._manage_client: dict[str, ManageClient] = dict()
         self._latest_manage_client_id = None
 
-    def _session_initial_set(self):
-        self._api_session = ChzzkAPISession(loop=self.loop)
-        self._game_session = NaverGameAPISession(loop=self.loop)
+        if not isinstance(self.loop, _LoopSentinel):
+            self._session_initial_set()
+
+    @staticmethod
+    def initial_async_setup(func):
+        @wraps(func)
+        async def wrapper(self: Self, *args, **kwargs):
+            if isinstance(self.loop, _LoopSentinel):
+                await self._async_setup_hook()
+            return await func(self, *args, **kwargs)
+
+        return wrapper
+
+    async def _async_setup_hook(self) -> None:
+        self.loop = loop = asyncio.get_running_loop()
+        self._session_initial_set(loop)
+
+    def _session_initial_set(self, loop: Optional[asyncio.AbstractEventLoop] = None) -> None:
+        self._api_session = ChzzkAPISession(loop=self.loop or loop)
+        self._game_session = NaverGameAPISession(loop=self.loop or loop)
+
+        if self.__authorization_key is not None and self.__session_key is not None:
+            self.login(self.__authorization_key, self.__session_key)
+
+        for manage_client in self._manage_client.values():
+            manage_client._session_initial_set(self.loop or loop)
 
     async def __aenter__(self) -> Self:
+        await self._async_setup_hook()
         return self
 
     async def __aexit__(
@@ -86,8 +112,9 @@ class Client:
     async def close(self):
         """Closes the connection to chzzk."""
         self._closed = True
-        await self._api_session.close()
-        await self._game_session.close()
+        if self._api_session is not None and self._game_session is not None:
+            await self._api_session.close()
+            await self._game_session.close()
 
         for manage_client in self._manage_client.values():
             if manage_client.is_closed:
@@ -107,13 +134,21 @@ class Client:
         session_key : str
             A `NID_SES` value in the cookie.
         """
+        if self._api_session is None or self._game_session is None:
+            self.__authorization_key = authorization_key
+            self.__session_key = session_key
+            return
+
         self._api_session.login(authorization_key, session_key)
         self._game_session.login(authorization_key, session_key)
 
     @property
     def has_login(self) -> bool:
+        if self._api_session is None or self._game_session is None:
+            return all((self.__authorization_key is not None, self.__session_key is not None))
         return all((self._api_session.has_login, self._game_session.has_login))
 
+    @initial_async_setup
     async def live_status(self, channel_id: str) -> Optional[LiveStatus]:
         """Get a live status info of broadcaster.
 
@@ -130,6 +165,7 @@ class Client:
         res = await self._api_session.live_status(channel_id=channel_id)
         return res.content
 
+    @initial_async_setup
     async def live_detail(self, channel_id: str) -> Optional[LiveDetail]:
         """Get a live detail info of broadcaster.
 
@@ -146,6 +182,7 @@ class Client:
         res = await self._api_session.live_detail(channel_id=channel_id)
         return res.content
 
+    @initial_async_setup
     async def user(self) -> User:
         """Get my user info.
         This method should be used after login.
@@ -158,6 +195,7 @@ class Client:
         res = await self._game_session.user()
         return res.content
 
+    @initial_async_setup
     async def search_channel(self, keyword: str) -> list[Channel]:
         """Search the channel with keyword.
 
@@ -175,6 +213,7 @@ class Client:
         data = res.content.data
         return [x.channel for x in data]
 
+    @initial_async_setup
     async def search_video(self, keyword: str) -> list[Video]:
         """Search the video with keyword.
 
@@ -197,6 +236,7 @@ class Client:
 
         return [x.video for x in data]
 
+    @initial_async_setup
     async def search_live(self, keyword: str) -> list[Live]:
         """Search the live with keyword.
 
@@ -219,8 +259,9 @@ class Client:
 
         return [x.live for x in data]
 
+    @initial_async_setup
     async def autocomplete(self, keyword: str) -> list[str]:
-        """Get a auto-completed keyword.
+        """Get an auto-completed keyword.
 
         Parameters
         ----------

--- a/chzzkpy/unofficial/client.py
+++ b/chzzkpy/unofficial/client.py
@@ -81,7 +81,9 @@ class Client:
         self.loop = loop = asyncio.get_running_loop()
         self._session_initial_set(loop)
 
-    def _session_initial_set(self, loop: Optional[asyncio.AbstractEventLoop] = None) -> None:
+    def _session_initial_set(
+        self, loop: Optional[asyncio.AbstractEventLoop] = None
+    ) -> None:
         self._api_session = ChzzkAPISession(loop=self.loop or loop)
         self._game_session = NaverGameAPISession(loop=self.loop or loop)
 
@@ -145,7 +147,9 @@ class Client:
     @property
     def has_login(self) -> bool:
         if self._api_session is None or self._game_session is None:
-            return all((self.__authorization_key is not None, self.__session_key is not None))
+            return all(
+                (self.__authorization_key is not None, self.__session_key is not None)
+            )
         return all((self._api_session.has_login, self._game_session.has_login))
 
     @initial_async_setup

--- a/chzzkpy/unofficial/manage/manage_client.py
+++ b/chzzkpy/unofficial/manage/manage_client.py
@@ -74,7 +74,9 @@ class ManageClient:
 
         return wrapper
 
-    def _session_initial_set(self, loop: Optional[asyncio.AbstractEventLoop] = None) -> None:
+    def _session_initial_set(
+        self, loop: Optional[asyncio.AbstractEventLoop] = None
+    ) -> None:
         self._http = ChzzkManageSession(self.client.loop or loop)
         self._http.login(
             authorization_key=self.client._api_session._authorization_key,

--- a/chzzkpy/unofficial/manage/manage_client.py
+++ b/chzzkpy/unofficial/manage/manage_client.py
@@ -23,7 +23,10 @@ SOFTWARE.
 
 from __future__ import annotations
 
-from typing import List, Literal, Optional, TYPE_CHECKING
+import asyncio
+
+from typing import List, Self, Literal, Optional, TYPE_CHECKING
+from functools import wraps
 from ..error import LoginRequired
 from ..user import PartialUser, UserRole
 from .enums import SortType, SubscriberTier
@@ -52,20 +55,37 @@ class ManageClient:
         self.client = client
 
         # All manage feature needs login.
-        if not self.client._api_session.has_login:
+        if not self.client.has_login:
             raise LoginRequired()
 
-        self._http = ChzzkManageSession(self.client.loop)
+        self._http: Optional[ChzzkManageSession] = None
+        if isinstance(self.client.loop, asyncio.AbstractEventLoop):
+            self._session_initial_set()
+
+        self._is_closed = False
+
+    @staticmethod
+    def initial_async_setup(func):
+        @wraps(func)
+        async def wrapper(self: Self, *args, **kwargs):
+            if not isinstance(self.client.loop, asyncio.AbstractEventLoop):
+                await self.client._async_setup_hook()
+            return await func(self, *args, **kwargs)
+
+        return wrapper
+
+    def _session_initial_set(self, loop: Optional[asyncio.AbstractEventLoop] = None) -> None:
+        self._http = ChzzkManageSession(self.client.loop or loop)
         self._http.login(
             authorization_key=self.client._api_session._authorization_key,
             session_key=self.client._api_session._session_key,
         )
-        self._is_closed = False
 
     async def close(self):
         """Closes the connection to chzzk."""
         self._is_closed = True
-        await self._http.close()
+        if self._http is not None:
+            await self._http.close()
         return
 
     @property
@@ -73,6 +93,7 @@ class ManageClient:
         """Indicates if the session is closed."""
         return self._is_closed
 
+    @initial_async_setup
     async def get_prohibit_words(self) -> List[ProhibitWord]:
         """Get prohibit words in chat.
 
@@ -87,6 +108,7 @@ class ManageClient:
         ]
         return prohibit_words
 
+    @initial_async_setup
     async def get_prohbit_word(self, word: str) -> Optional[ProhibitWord]:
         """Get prohibit word with word.
         When word does not contain prohibit word, returns None.
@@ -107,6 +129,7 @@ class ManageClient:
             return
         return prohibit_words[0]
 
+    @initial_async_setup
     async def add_prohibit_word(self, word: str) -> Optional[ProhibitWord]:
         """Add a prohibit word at chat.
 
@@ -123,6 +146,7 @@ class ManageClient:
         await self._http.add_prohibit_word(self.channel_id, word)
         return await self.get_prohbit_word(word)
 
+    @initial_async_setup
     async def edit_prohibit_word(
         self, prohibit_word: ProhibitWord | int, word: str
     ) -> Optional[ProhibitWord]:
@@ -149,6 +173,7 @@ class ManageClient:
         await self._http.edit_prohibit_word(self.channel_id, prohibit_word_number, word)
         return await self.get_prohbit_word(word)
 
+    @initial_async_setup
     async def remove_prohibit_word(self, prohibit_word: ProhibitWord | int) -> None:
         """Remove a prohibit word.
 
@@ -165,10 +190,12 @@ class ManageClient:
 
         await self._http.remove_prohibit_word(self.channel_id, prohibit_word_number)
 
+    @initial_async_setup
     async def remove_prohibit_words(self) -> None:
         """Remove all prohibit words."""
         await self._http.remove_prohibit_word_all(self.channel_id)
 
+    @initial_async_setup
     async def get_chat_rule(self) -> str:
         """Get chat rule of broadcast.
 
@@ -180,6 +207,7 @@ class ManageClient:
         data = await self._http.get_chat_rule(self.channel_id)
         return data.content.rule
 
+    @initial_async_setup
     async def set_chat_rule(self, word: str) -> None:
         """Set chat rule of broadcast.
 
@@ -190,6 +218,7 @@ class ManageClient:
         """
         await self._http.set_chat_rule(self.channel_id, word)
 
+    @initial_async_setup
     async def stream(self) -> Stream:
         """Get a stream key required for streamming.
 
@@ -201,6 +230,7 @@ class ManageClient:
         data = await self._http.stream(channel_id=self.channel_id)
         return data.content
 
+    @initial_async_setup
     async def add_restrict(
         self,
         user: str | PartialUser,
@@ -246,6 +276,7 @@ class ManageClient:
         return user
         return data.content
 
+    @initial_async_setup
     async def edit_restrict(
         self,
         user: str | PartialUser,
@@ -284,6 +315,7 @@ class ManageClient:
         user._set_manage_client(self)
         return user
 
+    @initial_async_setup
     async def remove_restrict(self, user: str | PartialUser) -> None:
         """Remove an user to restrict activity.
 
@@ -306,6 +338,7 @@ class ManageClient:
             channel_id=self.channel_id, target_id=target_id
         )
 
+    @initial_async_setup
     async def add_role(self, user: str | PartialUser, role: UserRole) -> PartialUser:
         """Add a broadcast permission to user.
 
@@ -341,6 +374,7 @@ class ManageClient:
         user._set_manage_client(self)
         return user
 
+    @initial_async_setup
     async def remove_role(self, user: str | PartialUser) -> None:
         """Remove a broadcast permission to user.
 
@@ -356,6 +390,7 @@ class ManageClient:
 
         await self._http.remove_role(channel_id=self.channel_id, target_id=user_id)
 
+    @initial_async_setup
     async def chat_activity_count(self, user: str | PartialUser) -> ChatActivityCount:
         """Get chat activity count of user.
 
@@ -380,7 +415,8 @@ class ManageClient:
         )
         return data.content
 
-    async def subcribers(
+    @initial_async_setup
+    async def subscribers(
         self,
         page: int = 0,
         size: int = 50,
@@ -389,7 +425,7 @@ class ManageClient:
         tier: Optional[SubscriberTier] = None,
         nickname: Optional[str] = None,
     ) -> ManageResult[ManageSubcriber]:
-        """Get subcribers of channel.
+        """Get subscribers of channel.
 
         Parameters
         ----------
@@ -422,6 +458,7 @@ class ManageClient:
         )
         return data.content
 
+    @initial_async_setup
     async def followers(
         self, page: int = 0, size: int = 50, sort_type: SortType = SortType.recent
     ) -> ManageResult[ManageFollower]:
@@ -448,6 +485,7 @@ class ManageClient:
             followed_user.user._set_manage_client(self)
         return data.content
 
+    @initial_async_setup
     async def restrict(
         self, page: int = 0, size: int = 50, nickname: Optional[str] = None
     ) -> ManageResult[UnrestrictRequest]:
@@ -477,6 +515,7 @@ class ManageClient:
             restricted_user._set_manage_client(self)
         return data.content
 
+    @initial_async_setup
     async def unrestrict_requests(
         self, page: int = 0, size: int = 50, nickname: Optional[str] = None
     ) -> ManageResult[UnrestrictRequest]:
@@ -506,6 +545,7 @@ class ManageClient:
             unrestrict_request._set_manage_client(self)
         return data.content
 
+    @initial_async_setup
     async def live_replay(
         self, page: int = 0, size: int = 50
     ) -> ManageResult[ManageVideo]:
@@ -528,6 +568,7 @@ class ManageClient:
         )
         return data.content
 
+    @initial_async_setup
     async def videos(self, page: int = 0, size: int = 50) -> ManageResult[ManageVideo]:
         """Get uploaded video of channel.
 

